core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3053,6 +3053,17 @@ public static double acos(double b0) {
     return Math.acos(b0);
   }
 
+  // ACOSD
+  /** SQL <code>ACOSD</code> operator applied to BigDecimal values. */
+  public static double acosd(BigDecimal b0) {
+    return Math.toDegrees(Math.acos(b0.doubleValue()));
```

Comment:
acos will supposedly throw if the value is out of range.
But the value of decimal can also be out of range for double, which will cause doubleValue to throw, and you will get a different error depending on its value.

Reply:
I have reworked the logic for these functions to try to better match the behaviour of the radian versions. The summary is:
ACOSD - returns NaN if the value is NaN or outside of -1..1
ASIND - returns NaN if the value is NaN or outside of -1..1
ATAND - returns NaN if the value is NaN or greater than `Double.MAX_VALUE`
COSD - returns NaN if the value is NaN or greater than `Double.MAX_VALUE`
SIND - returns NaN if the value is NaN or greater than `Double.MAX_VALUE`
TAND - returns NaN if the value is NaN or greater than `Double.MAX_VALUE`

When converting a `BigDecimal` value to a double, if it is larger than `Double.MAX_VALUE` then it becomes `Double.Infinity` and will be handled in the cases above.

Reply:
Is this what postgres does?

Reply:
I did some testing and no this doesn't match PostgreSQL. I have updated to match the behaviour I observed in PostgreSQL.

* ACOSD - returns NaN if input was NaN. Fails if outside of -1..1
* ASIND - returns NaN if input was NaN. Fails if outside of -1..1
* ATAND - returns NaN if input was NaN.
* COSD - returns NaN if input was NaN. Fails if input is greater than max double.
* SIND - returns NaN if input was NaN. Fails if input is greater than max double.
* TAND - returns NaN if input was NaN. Fails if input is greater than max double.